File: src/main/java/nl/asdproject2/GreetingClient.java
package nl.asdproject2;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.Socket;

public class GreetingClient
{
    private String serverName;
    private int port;
    private Socket client;
    private String username;

    public GreetingClient(String serverName, int port)
    {
        this.serverName = serverName;
        this.port = port;

        try
        {
            this.username = getTheUsername();
        }
        catch (IOException e)
        {
            e.printStackTrace();
        }
    }

    public void run()
    {
        try
        {
            Writer.write("Connecting to " + serverName + " on port " + port);
            client = new Socket(serverName, port);
            Writer.write("Just connected to " + client.getRemoteSocketAddress());

            new ClientInputThread(client.getInputStream()).start();

            startMessaging();
        }
        catch (IOException e)
        {
            e.printStackTrace();
        }
    }

    private void startMessaging()
    {
        String message = "";
        while (!message.toLowerCase().equals("close"))
        {
            try
            {
                message = input();
                Writer.writeUTF(client.getOutputStream(), username + ": " + message);
            }
            catch (IOException e)
            {
                e.printStackTrace();
            }
        }
        close();
    }

    private void close()
    {
        try
        {
            client.close();
        }
        catch (IOException e)
        {
            e.printStackTrace();
        }
    }

    private String input() throws IOException
    {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        return reader.readLine();
    }

    private String getTheUsername() throws IOException
    {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        System.out.print("Enter your username: ");
        return reader.readLine();
    }
}

File: src/main/java/nl/asdproject2/kotlin/GreetingClient.kt
package nl.asdproject2.kotlin

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.net.Socket

class GreetingClient(private val serverName: String, private val port: Int) {
    private var client: Socket? = null
    private var username: String? = null

    private val theUsername: String
        @Throws(IOException::class)
        get() {
            val reader = BufferedReader(InputStreamReader(System.`in`))
            print("Enter your username: ")
            return reader.readLine()
        }

    init {

        try {
            this.username = theUsername
        } catch (e: IOException) {
            e.printStackTrace()
        }

    }

    fun run() {
        try {
            Writer.write("Connecting to $serverName on port $port")
            client = Socket(serverName, port)
            Writer.write("Just connected to " + client!!.remoteSocketAddress)

            ClientInputThread(client!!.getInputStream()).start()

            startMessaging()
        } catch (e: IOException) {
            e.printStackTrace()
        }

    }

    private fun startMessaging() {
        var message = ""
        while (message.toLowerCase() != "close") {
            try {
                message = input()
                Writer.writeUTF(client!!.getOutputStream(), username + ": " + message)
            } catch (e: IOException) {
                e.printStackTrace()
            }

        }
        close()
    }

    private fun close() {
        try {
            client!!.close()
        } catch (e: IOException) {
            e.printStackTrace()
        }

    }

    @Throws(IOException::class)
    private fun input(): String {
        val reader = BufferedReader(InputStreamReader(System.`in`))
        return reader.readLine()
    }
}